lib/endpoints/class-wp-json-posts-controller.php

```diff
@@ -1124,6 +1124,15 @@ protected function prepare_links( $post ) {
 			);
 		}
 
+		if ( 'attachment' !== $post->post_type ) {
```

Comment:
@danielbachhuber Core doesn't have an explicit way to note if a post supports attachments.  A non-explict option would be to check if a post_type supports either `thumbnail` OR `editor`.